README.md

```diff
@@ -93,72 +73,156 @@ npm install
 
 ## Usage
 
-### Generating a new key pair
+### Installing Support for Key Types
 
-Ed25519:
+In order to use this library, you will need to import and install driver
+libraries for key types you'll be working with via the `use()` method.
+
+For example, to use this library with only the `ed25519` key type:
 
 ```js
-const {Ed25519KeyPair} = require('crypto-ld');
+import {CryptoLD} from 'crypto-ld';
+const cryptoLd = new CryptoLD();
+
+cryptoLd.use(require('ed25519-key-pair'));
 
-const keyPair = await Ed25519KeyPair.generate();
+// With only one key type installed, you do not need to specify key type for
+// most operations
+const keyPair = await cryptoLd.generate(); // generates an ed25519 key pair
 ```
 
-RSA:
+To use the library with all supported key types:
 
 ```js
-const {RSAKeyPair} = require('crypto-ld');
-
-const keyPair = await RSAKeyPair.generate();
+import Ed25519 from 'ed25519-key-pair';
+import RSA from 'rsa-key-pair';
+import Secp256k1 from 'secp256k1-key-pair';
+import X25519 from 'x25519-key-pair';
+
+import {CryptoLD} from 'crypto-ld';
+const cryptoLd = new CryptoLD();
+
+cryptoLd.use(Ed25519); // ed25519 type
```

Comment:
It's not clear to me how `crypto-ld` wants to handle two instances of the same key type but different cryptographic suites. We should make a recommendation for library authors creating libraries for new key types. Based on this pattern it seems to me that the library's default export should be its recommended/default cryptographic suite. The library should also have named exports for every cryptographic suite it supports. 

An example:
```js
// import recommended cryptographic suites
import Ed25519 from 'ed25519-key-pair';
// import specific cryptographic suites
import {Ed25519VerificationKey2018, Ed25519VerificationKey2020} from 'ed25519-key-pair';

```

Reply:
@gannan08 the idea is that we do /not/ handle two instances of the same key type.

Reply:
@dmitrizagidulin  It may be that I am still not getting something that you may understand as trivial knowledge. Let's assume that one has a ledger. That ledger is developed using `crypto-ld` and the `ed25519-key-pair` library. Soon thereafter, the ledger is launched and uses the `Ed25519VerificationKey2018` as its premiere cryptographic suite. After a few years, `Ed25519VerificationKey2020` is promoted as the recommended cryptographic suite for ED25519 keys.

The ledger wants to upgrade to the recommended cryptographic suite `Ed25519VerificationKey2020`.

This means that the ledger must continue using `Ed25519VerificationKey2018` to verify old signatures and use `Ed25519VerificationKey2020` to sign and verify new operations.

How is this achieved?

@dlongley @mattcollier Does this make sense? Am I off the mark?

Reply:
@dmitrizagidulin, @gannan08's point is valid, can you show how this module can be used for that use case?

Reply:
@dlongley @gannan08 I think we need a quick design conversation on how we're handling backwards-compatibility on our crypto suites.

Reply:
@dlongley @gannan08  I think I understand your concerns.

It sounds to me like we have the following design constraints:

1. For new/greenfield projects, we want the library+key type to just recommend the latest suite. So that, if a developer just knows they want an `ed25519` key type, then the latest suite gets automatically picked.
2. For projects that live through a suite change (and are able to use the newest suite), we want the code to still be backwards-compatible with the older suites, even if new keys are generated with the new one. 
3. Although we want to be backwards-compatible, at some point, projects may want to specifically exclude old/obsolete suites.
4. For projects that cannot upgrade suites for whatever reason, they must be able to "pin" all operations to one or more past suites.
4. As much as possible, we must shield developers from having to know individual suite names. (It's hard enough to explain to developers the difference between ed25519 keys and RSA keys; I feel like they should only be exposed to individual suite/year names only at the last resort.)

Are these constraints accurate? (Specifically, do we actually need 3 and 4?) 
Need to ponder what sort of API / export signature would satisfy all those.

Reply:
> For new/greenfield projects, we want the library+key type to just recommend the latest suite. So that, if a developer just knows they want an ed25519 key type, then the latest suite gets automatically picked.
> 
> As much as possible, we must shield developers from having to know individual suite names. (It's hard enough to explain to developers the difference between ed25519 keys and RSA keys; I feel like they should only be exposed to individual suite/year names only at the last resort.)

One of the reasons for suites is to surface those to users instead of key types. In fact, a goal for suites is to eventually just say "TheBestOne2020" -- and that would be all the user needs to know vs. esoteric key type information. We want to drive usage in this direction even if we have to support some helper mapping functions in the near term.

I agree with the other comments generally.

Reply:
Resolving this thread. On previous engineering call, the consensus was - we only specify the suits explicitly, no defaults or helpers, like so:

```js
import {Ed25519VerificationKey2018} from 'ed25519-verification-key-2018';

cryptoLd.use(Ed25519VerificationKey2018);
```